### PR TITLE
Switched to AsyncKeyedLock for better performance and less allocations.

### PR DESCRIPTION
**Pull request by MarkCiliaVincenti**

### CODE REVIEW

File: src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs [hunk 1]
```diff
@@ -12,116 +12,61 @@ namespace Laraue.Core.Threading
...
+        private readonly AsyncKeyedLocker<TKey> _asyncKeyedLocker;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="KeyedSemaphoreSlim{TKey}"/> class, specifying
         /// the initial and maximum number of requests that can be granted concurrently.
         /// </summary>
         /// <param name="initCount"></param>
         /// <param name="maxCount"></param>
+        [Obsolete("Passing on an initCount different to maxCount could have undesired results. Only pass on initCount instead.")]
         public KeyedSemaphoreSlim(int initCount, int maxCount)
```
Comment (win7user10) [inline src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs]: Could you please drop this constructor?

File: src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs [hunk 2]
```diff
@@ -12,116 +12,61 @@ namespace Laraue.Core.Threading
...
+            });
         }
         
         /// <summary>
         /// Initializes a new instance of the <see cref="KeyedSemaphoreSlim{TKey}"/> class, specifying
         /// the initial number of requests that can be granted concurrently.
         /// </summary>
         /// <param name="initCount"></param>
-        public KeyedSemaphoreSlim(int initCount) : this(initCount, initCount)
+        public KeyedSemaphoreSlim(int initCount)
```
Comment (win7user10) [inline src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs]: And rename this parameter to MaxCount?

File: src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs (hunk 1 above)
Comment (MarkCiliaVincenti) [reply src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs]: done

File: src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs (hunk 2 above)
Comment (MarkCiliaVincenti) [reply src/Laraue.Core.Threading/KeyedSemaphoreSlim.cs]: done

**Comment by MarkCiliaVincenti** [comment]
@win7user10 applied changes as you requested.

**Comment by MarkCiliaVincenti** [comment]
@win7user10 thanks for the merge. Now I think you need to create a new package version so it can be referenced in the other projects?

**Comment by win7user10** [comment]
@MarkCiliaVincenti, you can change the nuget version in the corresponding .csproj and make a PR or I will increase that version when will work with that library next time myself.

**Comment by MarkCiliaVincenti** [comment]
https://github.com/win7user10/Laraue.Core/pull/4

**Pull request merged**